BraveWallet/Settings/CustomNetworkDetailsView.swift

```diff
@@ -0,0 +1,458 @@
+              .foregroundColor(Color(.braveOrange))
+          }
+        }
+        .disabled(networkStore.isAddingNewNetwork)
+      }
+      ToolbarItemGroup(placement: .cancellationAction) {
+        Button(action: {
+          presentationMode.dismiss()
+        }) {
+          Text(Strings.cancelButtonTitle)
+            .foregroundColor(Color(.braveOrange))
+        }
+      }
+    }
+    .background(
+      Color.clear
+        .alert(isPresented: $isDuplicateIdErrorPresented) {
+          Alert(
+            title: Text(""),
+            message: Text(Strings.Wallet.networkIdDuplicationErrMsg),
+            dismissButton: .default(Text(Strings.OKString))
+          )
+        }
+    )
+    .background(
+      Color.clear
+        .alert(isPresented: $isGenericErrorPresented) {
+          Alert(
+            title: Text(Strings.Wallet.failedToAddCustomNetworkErrorTitle),
+            message: Text(Strings.Wallet.failedToAddCustomNetworkErrorMessage),
+            dismissButton: .default(Text(Strings.OKString))
+          )
+        }
+    )
+  }
+  
+  private func validateAllFields() -> Bool {
+    if model.networkId.input.isEmpty {
+      model.networkId.error = Strings.Wallet.customNetworkEmptyErrMsg
+    }
+    model.networkName.error = model.networkName.input.isEmpty ? Strings.Wallet.customNetworkEmptyErrMsg : nil
+    model.networkSymbolName.error = model.networkSymbolName.input.isEmpty ? Strings.Wallet.customNetworkEmptyErrMsg : nil
+    model.networkSymbol.error = model.networkSymbol.input.isEmpty ? Strings.Wallet.customNetworkEmptyErrMsg : nil
+    if model.networkDecimals.input.isEmpty {
+      model.networkDecimals.error = Strings.Wallet.customNetworkEmptyErrMsg
+    }
+    if model.rpcUrls.first(where: { !$0.input.isEmpty && $0.error == nil }) == nil { // has no valid url
+      if let index = model.rpcUrls.firstIndex(where: { $0.input.isEmpty }) { // find the first empty entry
+        model.rpcUrls[index].error = Strings.Wallet.customNetworkEmptyErrMsg // set the empty err msg
+      }
+    }
+    
+    if model.networkId.error != nil
+        || model.networkName.error != nil
+        || model.networkSymbolName.error != nil
+        || model.networkSymbol.error != nil
+        || model.networkDecimals.error != nil
+        || model.rpcUrls.filter({ !$0.input.isEmpty && $0.error == nil }).isEmpty {
+      return false
+    }
+    
+    return true
+  }
+  
+  private func addCustomNetwork() {
+    guard validateAllFields() else { return }
+    
+    var chainIdInHex = ""
+    if let idValue = Int(model.networkId.input) {
+      chainIdInHex = "0x\(String(format: "%02X", idValue))"
+    }
+    // Check if input chain id already existed for non-edit mode
+    if !isEditMode,
+        networkStore.ethereumChains.contains(where: { $0.id == chainIdInHex }) {
+      isDuplicateIdErrorPresented = true
+      return
+    }
+    
+    let network: BraveWallet.EthereumChain = .init().then {
```

Comment:
BraveWallet.EthereumChain should have a memberwise initializer that you can use instead of `.init().then`